doc/rst/developer/bestPractices/TestSystem.rst

```diff
@@ -466,28 +637,178 @@ include a future for the issue tracked on the `GitHub issues page`_.
 
 .. _`GitHub issues page`: https://github.com/chapel-lang/chapel/issues
 
-.. _extensions:
 
-Planned Changes of Testing System
-=================================
+Tracking Current Failure Mode
++++++++++++++++++++++++++++++
+
+Sometimes a future will change its behavior, but not be resolved.  The future
+should be updated to continue to track the issue as much as possible - to alert
+developers when this happens, it is necessary to track not only the expected
+good output but also the output indicating the current failure.  This is done
+via a ``.bad`` file.  The contents of a ``.bad`` file are similar to a ``.good``
+file and should match the currently generated output of the test.
```

Comment:
Providing `.compopts` or `.execopts` results in the output file being named `<test_name>.1.exec.out.tmp` or something along those lines (`.1-2.` when both are used).  I don't know how useful going into those details are, but it might be worth calling that strategy out since the presence of `CHPL_DEVELOPER=true` in the environment will alter the output generated in some cases (so compiling the program on outside of the testing system might lead to frustration)